packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -104,19 +105,18 @@ function UncontrolledInnerBlocks( props ) {
 		getBlockSupport( name, '__experimentalLayout' ) ||
 		EMPTY_OBJECT;
 
-	const { allowSizingOnChildren = false } = defaultLayoutBlockSupport;
-	const usedLayout = layout || defaultLayoutBlockSupport;
-
 	const memoedLayout = useMemo(
 		() => ( {
-			// Default layout will know about any content/wide size defined by the theme.
+			// Default layout contains theme.json settings such as content and wide size.
 			...defaultLayout,
-			...usedLayout,
-			...( allowSizingOnChildren && {
-				allowSizingOnChildren: true,
-			} ),
+			// Any layout settings added to the block.
+			...layout,
+			// The layout settings from the block's block.json.
+			...defaultLayoutBlockSupport,
+			// Any alignment set on the block.
+			...( align && { alignWidth: align } ),
```

Comment:
I'm no guru in this area, but it makes sense to me to make inner blocks aware of their parents' alignment 👍🏻